# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: layout/painting/nsDisplayList.h
# Commit: bdb29ecb2b16
# Full Hash: bdb29ecb2b16458ca5c84594a5c9a0a6febc6249
# Author: Matt Woodrow <mwoodrow@mozilla.com>
# Date: 2021-05-16 09:17:48
# Regressor Bug: 1540737
# File Overlap Count: 1
# Description:
#   Bug 1540737 - Allow filter painting to take a callback for painting children. r=miko
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D114670
# ==============================================================================

diff -r f54a65e66247 -r bdb29ecb2b16 layout/painting/nsDisplayList.h
--- a/layout/painting/nsDisplayList.h	Sat May 15 22:16:48 2021 +0000
+++ b/layout/painting/nsDisplayList.h	Sat May 15 22:38:43 2021 +0000
@@ -6649,6 +6649,10 @@
   void PaintAsLayer(nsDisplayListBuilder* aBuilder, gfxContext* aCtx,
                     LayerManager* aManager);
 
+  void PaintWithContentsPaintCallback(
+      nsDisplayListBuilder* aBuilder, gfxContext* aCtx,
+      const std::function<void(gfxContext* aContext)>& aPaintChildren);
+
   bool CreateWebRenderCommands(
       mozilla::wr::DisplayListBuilder& aBuilder,
       mozilla::wr::IpcResourceUpdateQueue& aResources,